bot/exts/fun/guess_number.py

```diff
@@ -0,0 +1,102 @@
+import random
+
+from discord import Message
+from discord.ext import commands
+
+from bot.bot import Bot
+
+
+class Difficulty(commands.Converter):
+    """Class used to convert the `difficulty` parameter."""
+
+    EASY = ("e", "eas", "easy")
+    MEDIUM = ("m", "med", ",medium")
+    HARD = ("h", "hard")
+
+    async def convert(self, ctx: commands.Context, diff: str) -> str:
+        """Converts the provided `difficulty` into the corresponding string."""
+        if diff.lower() in self.EASY:
+            return "easy"
+
+        if diff.lower() in self.MEDIUM:
+            return "medium"
+
+        if diff.lower() in self.HARD:
+            return "hard"
+
+        raise commands.BadArgument(f"{diff!r} is not a valid difficulty.")
+
+
+class GuessNumber(commands.Cog):
+    """Cog for the GuessNumber command."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        self.active_games = {}
+
+    @commands.command(name="guessnumber", aliases=("gn", "numbergame"))
+    async def guess_number(self, ctx: commands.Context, difficulty: Difficulty = "medium") -> None:
+        """Starts a new game of guessing the number with default difficulty set to medium."""
+        # Duplicate game check
+        if ctx.author.id in self.active_games:
+            await ctx.send("You already have an active game. Finish that one before starting a new one.")
+            return
+
+        if difficulty == "easy":
```

Comment:
if someone guesses randomly then yes, but if you think logically each guess the number can be halved.
First Guess: Pick the middle number, 25.
If the correct number is lower, you now have a range of 1-24.
If the correct number is higher, you now have a range of 26-50.
Second Guess: Again, pick the middle number of the narrowed range. and so on.

So, the chances would be like:-
Guess 1: 1 in 50 chance of being correct.
Guess 2: 1 in 25 chance of being correct.
Guess 3: 1 in 12 chance of being correct.
Guess 4: 1 in 6 chance of being correct.
Guess 5: 1 in 3 chance of being correct.

same logic applies fro medium and hard difficulty, the range is increased tho.
lmk if you still think it's too high and want to decrease the range.

Reply:
Sure, but fun games where a solution is an easy rational one (by using binary search), isn't fun anymore. 
I wouldn't play it like this anyway, we'll see what others have to say.

Reply:
I agree that 1-50 might be a bit hard for easy.

I personally feel that 1-10 is suitable for easy, medium being 1-50 and hard 1-100 or 1-200.

This is actually a good game to teach binary search. Using 1-10 for the easy level makes it so that it's a certain win if binary search is used, and there is still a decent chance of winning if the player is only randomly guessing.